scripts/chgres_cube.py

```diff
@@ -0,0 +1,286 @@
+            logging.error(f"Value at {pathstr} must be a dictionary")
+            sys.exit(1)
+        config = subconfig
+    return config
+
+
+def parse_args(argv):
+    """
+    Parse arguments for the script.
+    """
+    parser = ArgumentParser(
+        description="Script that runs chgres_cube via uwtools API",
+    )
+    parser.add_argument(
+        "-c",
+        "--config-file",
+        metavar="PATH",
+        required=True,
+        help="Path to experiment config file.",
+        type=Path,
+    )
+    parser.add_argument(
+        "--cycle",
+        help="The cycle in ISO8601 format (e.g. 2024-07-15T18)",
+        required=True,
+        type=dt.datetime.fromisoformat,
+    )
+    parser.add_argument(
+        "--key-path",
+        help="Dot-separated path of keys leading through the config to the driver's YAML block",
+        metavar="KEY[.KEY...]",
+        required=True,
+        type=lambda s: s.split("."),
+    )
+    parser.add_argument(
+        "--member",
+        default="000",
+        help="The 3-digit ensemble member number.",
+    )
+    return parser.parse_args(argv)
+
+
+# pylint: disable=too-many-locals, too-many-statements, too-many-branches
+def run_chgres_cube(config_file, cycle, key_path, member):
+    """
+    Setup and run the chgres_cube Driver.
+    """
+
+    # dereference expressions during driver initialization
+    expt_config = get_yaml_config(config_file)
+    CRES = expt_config["workflow"]["CRES"]
+    os.environ["CRES"] = CRES
+    os.environ["MEMBER"] = member
+
+    # set universal variables
+    cyc = str(expt_config["workflow"]["DATE_FIRST_CYCL"])[8:10]
+    dot_ensmem = (
+        f".mem{member}"
+        if (
+            expt_config["user"]["RUN_ENVIR"] == "nco"
+            and expt_config["global"]["DO_ENSEMBLE"]
+            and member
+        )
+        else ""
+    )
+    nco_net = expt_config["nco"]["NET_default"]
+
+    # Extract driver config from experiment config
+    chgres_cube_driver = ChgresCube(
+        config=config_file,
+        cycle=cycle,
+        key_path=key_path,
+    )
+    rundir = Path(chgres_cube_driver.config["rundir"])
+    print(f"Will run in {rundir}")
+
+    # Dereference cycle for file paths
+    expt_config_cp = get_yaml_config(deepcopy(expt_config.data))
+    expt_config_cp.dereference(
+        context={
+            "cycle": cycle,
+            **expt_config_cp,
+        }
+    )
+    chgres_cube_config = _walk_key_path(expt_config_cp, key_path)
+    input_type = chgres_cube_config["chgres_cube"]["namelist"]["update_values"][
+        "config"
+    ].get("input_type")
+
+    # update config for ics task, run and stage data
+    if "task_make_ics" in key_path:
+        varsfilepath = chgres_cube_config["input_files_metadata_path"]
+        shconfig = _parse_var_defns(varsfilepath)
+        extrn_config_fns = shconfig["EXTRN_MDL_FNS"]
+        extrn_config_fhrs = shconfig["EXTRN_MDL_FHRS"]
+
+        if input_type == "grib2":
```

Comment:
Hm - if we move everything out it might be ok. I was getting errors in testing where the YAML validator was unhappy about blank path keys, but if it has that it might be ok. Certainly worth trying!